platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -59,169 +66,91 @@
  *   <li>https.protocols: Secure protocols supported by the Solr server
  *   <li>https.cipherSuites: Cipher suites supported by the Solr server
  * </ul>
+ *
+ * @deprecated This class may be removed in the future
  */
 @Deprecated
 public final class HttpSolrClientFactory implements SolrClientFactory {
-  private static final String HTTPS_PROTOCOLS = "https.protocols";
+
+  public static final String DEFAULT_SCHEMA_XML = "schema.xml";
+  public static final String DEFAULT_SOLRCONFIG_XML = "solrconfig.xml";
   private static final String HTTPS_CIPHER_SUITES = "https.cipherSuites";
-  private static final String SOLR_CONTEXT = "/solr";
+  private static final String HTTPS_PROTOCOLS = "https.protocols";
+  private static final String KEY_STORE = "javax.net.ssl.keyStore";
+  private static final String KEY_STORE_PASS = "javax.net.ssl.keyStorePassword";
+  public static final String KEY_STORE_TYPE = "javax.net.ssl.keyStoreType";
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
+  public static final String SOLR_CREDENTIALS = "solr.credentials";
   private static final String SOLR_DATA_DIR = "solr.data.dir";
   private static final String SOLR_HTTP_URL = "solr.http.url";
-  private static final String KEY_STORE_PASS = "javax.net.ssl.keyStorePassword";
+  public static final String SOLR_USERNAME = "solr.username";
   private static final String TRUST_STORE = "javax.net.ssl.trustStore";
   private static final String TRUST_STORE_PASS = "javax.net.ssl.trustStorePassword";
-  private static final String KEY_STORE = "javax.net.ssl.keyStore";
-  public static final List<String> DEFAULT_PROTOCOLS;
-  public static final List<String> DEFAULT_CIPHER_SUITES;
-
-  static {
-    DEFAULT_PROTOCOLS =
-        AccessController.doPrivileged(
-            (PrivilegedAction<List<String>>)
-                () ->
-                    Collections.unmodifiableList(
-                        Arrays.asList(
-                            StringUtils.split(System.getProperty(HTTPS_PROTOCOLS, ""), ","))));
-    DEFAULT_CIPHER_SUITES =
-        AccessController.doPrivileged(
-            (PrivilegedAction<List<String>>)
-                () ->
-                    Collections.unmodifiableList(
-                        Arrays.asList(
-                            StringUtils.split(System.getProperty(HTTPS_CIPHER_SUITES, ""), ","))));
-  }
+  private final Map<String, String> propertyCache = new HashMap<>();
```

Comment:
Using a map to cache properties that are only use in a given invocation flow seems an overkill for something like 4ish properties. Just pass it around to the private methods as it used to be. This seems to me like we are overcomplicating it.